compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Numeric.hs

```diff
@@ -0,0 +1,105 @@
+-- Copyright (c) 2019 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- | DAML-LF Numeric literals, with scale attached.
+module DA.Daml.LF.Ast.Numeric
+    ( Numeric
+    , numeric
+    , numericScale
+    , numericMaxScale
+    ) where
+
+import Control.DeepSeq
+import Control.Monad
+import Data.Data
+import Data.Decimal
+import Data.Maybe
+import GHC.Generics (Generic)
+import Numeric.Natural
+
+-- | Numeric literal. This must encode both the mantissa (up to 38 digits) and
+-- the scale (0-37), the latter controlling how many digits appear after the
+-- decimal point. Furthermore, when reading or writing a numeric instance,
```

Comment:
`instance` isn’t a typeclass instance right? I guess you mean a value of type `Numeric`?

Reply:
Nice catch! :)